tools/spec_to_manifest.py

```diff
@@ -0,0 +1,189 @@
+    # For now, keep as blank
+    manifest_obj["overview"] = ""
+
+    # Key Features -- mandatory
+    # From .md, for now, keep as blank
+    manifest_obj["key_features"] = []
+
+    # Requirements -- Optional
+    # Tricky reading from .md, due to different formatting....
+    # Keep blank for now
+    manifest_obj["requirements"] = []
+
+    # Resources -- optional
+    # Don't know where to pull this info from.....
+    manifest_obj["resources"] = []
+
+    # Version -- optional
+    # Directly from yaml, parsed as string in manifest
+    manifest_obj["version"] = str(yaml_obj.get("version"))
+
+    # Version History....
+    # Should be from .md -- need to parse
+    manifest_obj["version_history"] = []
+
+    # Publisher -- mandatory
+    # "Vendor" field from yaml
+    manifest_obj["publisher"] = yaml_obj.get("vendor")
+
+    # Support -- mandatory
+    # from yaml if R7, where to get info on "community" support?
+    if yaml_obj.get("support") == "rapid7":
+        support_obj = {"type": "publisher", "url": "http://www.rapid7.com/discuss"}
+    else:
+        support_obj = {"type": "community", "url": ""}
+    manifest_obj["support"] = support_obj
+
+    # Rapid7 Products -- mandatory
+    # Products key from yaml, we don't have an understanding of primary/secondary....
+    product_listings = yaml_obj.get("products")
+    r7_products = []
+    for x in range(len(product_listings)):
+        if x == 0:
+            r7_products.append({"name": product_listings[x], "role": "primary"})
+        else:
+            r7_products.append({"name": product_listings[x], "role": "secondary"})
+    manifest_obj["rapid7_products"] = r7_products
+
+    # Required R7 Features -- Required
+    # Different rules for workflows, plugins, etc....Workflows --> need orchestrator
+    manifest_obj["required_rapid7_features"] = ["orchestrator"]
+
+    # Status -- optional
+    # From yaml
+    manifest_obj["status"] = yaml_obj.get("status")
+
+    # Logos -- optional
+    # Not sure what this would mean for a workflow....
+    manifest_obj["logos"] = {}
+
+    # Display Options -- mandatory
+    # No sense of this in spec.....
+    manifest_obj["display_options"] = []
+
+    # Tags -- Mandatory
+    # Converting from yaml
+    hub_tags = yaml_obj.get("hub_tags")
+
+    # Tags.categories -- Mandatory
+    categories = hub_tags.get("use_cases")
+
+    # Tags.third_party_products -- optional
+    # Is this in tha YAML????
+    third_party_products = []
+
+    # Tags.keywords -- optional
+    keywords = hub_tags.get("keywords")
+    manifest_obj["tags"] = {"categories": categories, "third_party_products": third_party_products,
+                            "keywords": keywords}
+
+    # Documentation -- optional
+    # Not entirely sure how to handle this either, not in yaml
+    manifest_obj["documentation"] = []
+
+    # Media -- optional
+    # Small conversions from yaml
+    media_list = []
+    for media in yaml_obj.get("resources").get("screenshots"):
```

Comment:
We should test if screenshots exist, plugins for example do not have screenshots key in the spec and this code will fail with a KeyError or something similar here.